o3de/o3de
# Fix for Project Manager being able to use remote repo gems that include python tests

## What does this PR do?
When adding a remote repository in the the Project Manager, using a gem from that remote repository will fail to configure if it has any Python Tests defined as a target. In this case, the ROS2 gem includes [python smoke tests](https://github.com/o3de/o3de-extras/blob/development/Gems/ROS2/Code/PythonTests/CMakeLists.txt). However, the SDK build intentionally strips out all tests in its prepackaged gem. Since gems that come from external repos do not go through the SDK build process, they don't get stripped out, but this is addressed by baking in `LY_DISABLE_TEST_MODULES=ON` during the SDK build. There is an issue with Linux this property isn't propagated to the PAL_Linux.cmake properly, thus rendering the `PAL_TRAIT_BUILD_TESTS_SUPPORTED` gate to not exclude the test as intended. So when the project is configured, cmake reports the following error
```
-- Configuring done
CMake Error at /opt/O3DE/23.10.1/cmake/Dependencies.cmake:40 (add_dependencies):
  The dependency target "Legacy::Editor" of target "ROS2.PythonTests" does
  not exist.
Call Stack (most recent call first):
  /opt/O3DE/23.10.1/cmake/LYTestWrappers.cmake:246 (ly_add_dependencies)
  /opt/O3DE/23.10.1/cmake/LYTestWrappers.cmake:339 (ly_add_test)
  /home/o3de/O3DE/Gems/ROS2/2.0.0/Code/PythonTests/CMakeLists.txt:9 (ly_add_pytest)

-- Generating done
CMake Generate step failed.  Build files cannot be regenerated correctly.
```

This fix addresses that issue propogating the `PAL_TRAIT_BUILD_TESTS_SUPPORTED` property to the generated SDK for Linux. 


Fixes https://github.com/o3de/o3de/issues/16821 

## How was this PR tested?
Performed the following steps:

* Created a debian package
* Downloaded and installed debian package on a clean ubuntu 22.04 image (with ROS2 and O3DE prereq installed)
* Enable the ROS2 environment
* Launched project manager
* Added the stabilization remote repo ( https://raw.githubusercontent.com/o3de/o3de-extras/stabilization/2310 )
* Created new project based on the ROS template (make sure that the ROS2 and other necessary gems are downloaded and enabled)
* Build the project
* Open the Editor
